extensions/amp-video/0.1/test/test-video-cache.js

```diff
@@ -0,0 +1,159 @@
+      const videoEl = createVideo([{src: 'video1.mp4'}]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video1.mp4');
+    });
+
+    it('should select the first source if there are similar sources', () => {
+      const videoEl = createVideo([{src: 'video1.mp4'}, {src: 'video2.mp4'}]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video1.mp4');
+    });
+
+    it('should select the mp4 source if there are many sources', () => {
+      const videoEl = createVideo([
+        {src: 'video1.mp4'},
+        {src: 'video2.mp4', type: 'video/mp4'},
+      ]);
+      expect(videoCache.selectVideoSource(videoEl)).to.contain('video2.mp4');
+    });
+  });
+
+  describe('add sources', () => {
+    it('should set the correct attributes on the source added', () => {
+      const videoEl = createVideo([]);
+      videoCache.applySourcesToVideo(videoEl, [
+        {'url': 'video1.mp4', 'bitrate_kbps': 700, type: 'video/mp4'},
+      ]);
+      const addedSource = videoEl.querySelector('source');
+      expect(addedSource.getAttribute('src')).to.equal('video1.mp4');
+      expect(addedSource.getAttribute('data-bitrate')).to.equal('700');
+      expect(addedSource.getAttribute('type')).to.equal('video/mp4');
+    });
+
+    it('should add the sources sorted by bitrate', () => {
+      const videoEl = createVideo([]);
+      videoCache.applySourcesToVideo(videoEl, [
+        {'url': 'video1.mp4', 'bitrate_kbps': 700, type: 'video/mp4'},
+        {'url': 'video2.mp4', 'bitrate_kbps': 2000, type: 'video/mp4'},
+        {'url': 'video3.mp4', 'bitrate_kbps': 1500, type: 'video/mp4'},
+      ]);
+      const addedSources = videoEl.querySelectorAll('source');
+      expect(addedSources[0].getAttribute('data-bitrate')).to.equal('2000');
+      expect(addedSources[1].getAttribute('data-bitrate')).to.equal('1500');
+      expect(addedSources[2].getAttribute('data-bitrate')).to.equal('700');
+    });
+  });
+
+  describe('url forming', () => {
+    it('should send the request to the correct address if the video has an absolute url', async () => {
+      const videoEl = createVideo([{'src': 'https://website.com/video.html'}]);
+      const xhrSpy = env.sandbox.spy(xhrService, 'fetch');
+
+      const ampVideo = new AmpVideo(videoEl);
+      await ampVideo.buildCallback();
+
+      expect(xhrSpy).to.have.been.calledWith(
+        'https://website-com.cdn.ampproject.org/mbv/s/website.com/video.html'
+      );
+    });
+
+    it('should send the request to the correct address if the video has a relative url', async () => {
+      const videoEl = createVideo([{'src': 'video.html'}]);
+      const xhrSpy = env.sandbox.spy(xhrService, 'fetch');
+
+      const ampVideo = new AmpVideo(videoEl);
+      await ampVideo.buildCallback();
+
+      expect(xhrSpy).to.have.been.calledWith(
+        'https://example-com.cdn.ampproject.org/mbv/s/example.com/video.html'
+      );
+    });
+  });
+
+  describe('end to end', () => {
```

Comment:
If you want to write an amp-video test, it should live in amp-video.
Here, you should only test through the `fetchCachedSources` entry point